pkg/util/compile.go

```diff
@@ -0,0 +1,129 @@
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/Ne0nd0g/merlin/pkg/messages"
+)
+
+//ValidCompilerTargets is all the valid compiler targets? greetz 2 moloch--
+var ValidCompilerTargets = map[string]bool{
+	"darwin/386":      true,
+	"darwin/amd64":    true,
+	"dragonfly/amd64": true,
+	"freebsd/386":     true,
+	"freebsd/amd64":   true,
+	"freebsd/arm":     true,
+	"linux/386":       true,
+	"linux/amd64":     true,
+	"linux/arm":       true,
+	"linux/arm64":     true,
+	"linux/ppc64":     true,
+	"linux/ppc64le":   true,
+	"linux/mips":      true,
+	"linux/mipsle":    true,
+	"linux/mips64":    true,
+	"linux/mips64le":  true,
+	"linux/s390x":     true,
+	"netbsd/386":      true,
+	"netbsd/amd64":    true,
+	"netbsd/arm":      true,
+	"openbsd/386":     true,
+	"openbsd/amd64":   true,
+	"openbsd/arm":     true,
+	"plan9/386":       true,
+	"plan9/amd64":     true,
+	"plan9/arm":       true,
+	"solaris/amd64":   true,
+	"windows/386":     true,
+	"windows/amd64":   true,
+}
+
+//ValidGoos returns a list of valid Goos compiler targets
+func ValidGoos(string) []string {
+	return []string{
+		"darwin",
+		"dragonfly",
+		"freebsd",
+		"linux",
+		"netbsd",
+		"openbsd",
+		"plan9",
+		"solaris",
+		"windows",
+	}
+}
+
+//ValidGoarch returns a list of valid goarch compiler targets
+func ValidGoarch(string) []string {
+	return []string{
+		"386",
+		"amd64",
+		"arm",
+		"arm64",
+		"ppc64",
+		"ppc64le",
+		"mips",
+		"mipsle",
+		"mips64",
+		"mips64le",
+		"s390x",
+	}
+}
+
+//GoConfig is a configuration of values useful to the go compiler
+type GoConfig struct {
+	CGO    string
+	GOOS   string
+	GOARCH string
+	GOROOT string
+	GOPATH string
+}
+
+//GoCmd runs a command with the provided env config, in the specified directory
+func GoCmd(config GoConfig, cwd string, command []string) error {
```

Comment:
Do you think it would be valuable to put `GoCmd` into its own package incase future work could leverage the function outside of compiling?

Reply:
I'm not sure what else you would do with GoCmd other than compile. Happy to rename the `compile` file to something more generic (`localtools`?), but I think keeping GoCmd with the rest of it is the most sane way of doing it for now